cli/azd/pkg/input/console.go

```diff
@@ -591,31 +580,79 @@ func (c *AskerConsole) Handles() ConsoleHandles {
 	return c.handles
 }
 
-const cMinConsoleWidth int = 40
-
 func getConsoleWidth() int {
 	width, _ := consolesize.GetConsoleSize()
-	if width < cMinConsoleWidth {
-		return cMinConsoleWidth
-	}
 	return width
 }
 
+func (c *AskerConsole) handleResize(width int) {
+	c.consoleWidth.Store(int32(width))
+
+	c.spinnerLineMu.Lock()
+	if c.spinner.Status() == yacspin.SpinnerRunning {
+		line := c.spinnerLine(c.spinnerCurrentTitle, c.currentIndent.Load())
+		c.spinner.Message(line.Message)
+		_ = c.spinner.CharSet(line.CharSet)
+		c.spinner.Prefix(line.Prefix)
+	}
+	c.spinnerLineMu.Unlock()
+}
+
+func watchConsoleWidth(c *AskerConsole) {
+	if runtime.GOOS == "windows" {
+		go func() {
+			prevWidth := getConsoleWidth()
+			for {
+				time.Sleep(time.Millisecond * 250)
+				width := getConsoleWidth()
+
+				if prevWidth != width {
+					c.handleResize(width)
+				}
+				prevWidth = width
+			}
+		}()
+	} else {
+		// avoid taking a dependency on syscall.SIGWINCH (unix-only constant) directly
+		const SIGWINCH = syscall.Signal(0x1c)
+		signalChan := make(chan os.Signal, 1)
+		signal.Notify(signalChan, SIGWINCH)
+		go func() {
+			for range signalChan {
+				c.handleResize(getConsoleWidth())
+			}
+		}()
+	}
+}
+
 // Creates a new console with the specified writer, handles and formatter.
 func NewConsole(noPrompt bool, isTerminal bool, w io.Writer, handles ConsoleHandles, formatter output.Formatter) Console {
 	asker := NewAsker(noPrompt, isTerminal, handles.Stdout, handles.Stdin)
 
-	return &AskerConsole{
+	c := &AskerConsole{
 		asker:         asker,
 		handles:       handles,
 		defaultWriter: w,
 		writer:        w,
 		formatter:     formatter,
 		isTerminal:    isTerminal,
-		consoleWidth:  getConsoleWidth(),
-		initialWriter: w,
+		consoleWidth:  atomic.NewInt32(int32(getConsoleWidth())),
+		currentIndent: atomic.NewString(""),
 		noPrompt:      noPrompt,
 	}
+
+	spinnerConfig := yacspin.Config{
+		Frequency:    200 * time.Millisecond,
+		Writer:       c.writer,
+		Suffix:       " ",
+		TerminalMode: spinnerTerminalMode(isTerminal),
+		CharSet:      spinnerCharSet,
+	}
+	c.spinner, _ = yacspin.New(spinnerConfig)
```

Comment:
So we only have 1 spinner instance now instead of them getting created/destroyed during Show/Stop calls?
I would validate this is all working as expected in sceanrios where we pause/unpause spinner in cases where we start preview operation.  One area that comes to mind is when hooks run we pause and running spinner, show the preview experience and when the preview is complete we unpause the spinner.

Reply:
It does indeed work in pause/unpause scenarios with previewer -- for example Docker output in `azd package`.